File: features/identifier/src/main/java/com/nivasafinance/features/identifier/service/IdentifierService.java
package com.nivasafinance.features.identifier.service;

import com.nivasafinance.common.dto.IdentifierData;
import com.nivasafinance.common.dto.IdentifierRequest;


public interface IdentifierService {

    IdentifierData createIdentifierData(IdentifierRequest identifierRequest);
}

File: features/identifiers/src/main/java/com/nivasafinance/features/identifiers/service/IdentifierService.kt
package com.nivasafinance.features.identifiers.service

import com.nivasafinance.features.identifiers.dto.IdentifierRequest
import com.nivasafinance.features.identifiers.dto.IdentifierResponse
import com.nivasafinance.features.identifiers.dto.IdentifierUpdateRequest
import java.util.UUID

interface IdentifierService {
    fun createIdentifier(identifierRequest: IdentifierRequest): IdentifierResponse
    fun updateIdentifier(identifierId: UUID, identifierUpdateRequest: IdentifierUpdateRequest): IdentifierResponse
    fun deleteIdentifier(identifierId: UUID)
    fun getIdentifierById(identifierId: UUID): IdentifierResponse
    fun getAllIdentifiers(): List<IdentifierResponse>
}
